src/Compilers/CSharp/Portable/Emitter/Model/PEAssemblyBuilder.cs

```diff
@@ -56,9 +62,22 @@ internal abstract class PEAssemblyBuilderBase : PEModuleBuilder, Cci.IAssemblyRe
 
         public override ISourceAssemblySymbolInternal SourceAssemblyOpt => _sourceAssembly;
 
-        internal override ImmutableArray<NamedTypeSymbol> GetAdditionalTopLevelTypes()
+        internal override ImmutableArray<NamedTypeSymbol> GetAdditionalTopLevelTypes(DiagnosticBag diagnostics)
         {
-            return _additionalTypes;
+            var builder = ArrayBuilder<NamedTypeSymbol>.GetInstance();
+            builder.AddRange(_additionalTypes);
+
+            CreateEmbeddedAttributesIfNeeded(diagnostics);
```

Comment:
It seems reasonable to include embedded attributes in `GetAdditionalTopLevelTypes`. But is there a reasonable point where we could call `CreateEmbeddedAttributesIfNeeded` exactly once rather than generating diagnostics each time (or only the first time) from `GetAdditionalTopLevelTypes`?